roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -107,6 +107,108 @@ object rblstring {
     }
   }
 
+  /**
+    * Define the string-set-nth primitive.
+    * This replaces the n element of a String
+    *
+    * examples:
+    *  (string-set-nth "abcdef" 2 #\C) ===> "abCdef"
+    */
+  object stringSetNth extends Prim {
+    override val name: String = "string-set-nth"
+    override val minArgs: Int = 3
+    override val maxArgs: Int = 3
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+
+      for {
+        str <- checkRblString(0, elem)                      // Ensure arg0 is a RblString
+        n   <- checkFixnumBounds(1, elem, str.value.length) // Ensure arg1 is a Fixnum
+        ch  <- checkChar(2, elem)                           // Ensure arg2 is a Char
+      } yield {
+        RblString(str.value.updated(n, ch.value).toString)
+      }
+    }
+  }
+
+  /**
+    * Define the string-length primitive.
+    * Returns the length of the string
+    * e.g. (string-length "abcdefghi") ==> 9
+    */
+  object stringLength extends Prim {
+    override val name: String = "string-length"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 1
+
+    @checkTypeMismatch[RblString] // All args must be strings
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
+      val elem = ctxt.argvec.elem
+
+      Right(Fixnum(elem(0).asInstanceOf[RblString].value.length))
+    }
+  }
+
+  /**
+    * Define the string-new primitive.
+    * This creates a new RblString of length n. It is filled with
+    * either spaces ' ' or an optionally specified character.
+    *
+    * examples:
+    *  (string-new 5) ===> "     "
+    *  (string-new 4 #\A) ===> "AAAA"
+    */
+  object stringNew extends Prim {
+    override val name: String = "string-new"
+    override val minArgs: Int = 1
+    override val maxArgs: Int = 2
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem  = ctxt.argvec.elem
+      val nargs = ctxt.nargs
+
+      checkFixnum(0, elem).flatMap { n => // Ensure arg0 is a Fixnum
```

Comment:
You can use if statements in for expressions.

Reply:
What would that syntax look like?  I can't make it work because the existence of ctxt.argvec.elem(1) is optional. ch needs to be set to " " if it's not there.

Reply:
Done